Source/JavaScriptCore/wasm/WasmModuleInformation.h

```diff
@@ -137,6 +141,9 @@ struct ModuleInformation : public ThreadSafeRefCounted<ModuleInformation> {
 
     BitVector m_declaredFunctions;
     BitVector m_declaredExceptions;
+    BitVector m_simdFunctions;
+    BitVector m_seenFunctions;
```

Comment:
We should have flags in `FunctionData` and update it concurrently from threads without taking a lock. `functions[functionIndex]` can offer this data, and `functions` are already allocated when parsing a function. And since # of functions are declared before parsing functions, we are already allocating FunctionData for all functions, and Vector will not be resized. So you can access to FunctionData fully concurrently without taking a lock.